prov/verbs/src/ep_rdm/verbs_cq_ep_rdm.c

```diff
@@ -114,7 +112,7 @@ ssize_t fi_ibv_rdm_cq_sreadfrom(struct fid_cq *cq, void *buf, size_t count,
 				int timeout)
 {
 	size_t threshold = count;
-	uint64_t time_limit = fi_gettime_ms() + timeout;
+	uint64_t time_limit = ((timeout < 0) ? -1 : (fi_gettime_ms() + timeout));
```

Comment:
Shouldn't time_limit be assigned 0 when timeout < 0? The check on line 138 is !time_limit for endless loop.